pyro/infer/mcmc/mcmc.py

```diff
@@ -146,7 +146,7 @@ def _traces(self, *args, **kwargs):
                 w.start()
             while active_workers:
                 try:
-                    chain_id, val = self.result_queue.get_nowait()
+                    chain_id, val = self.result_queue.get(timeout=5)
```

Comment:
I wonder if this will make any difference?

Reply:
We are blocking and not infinitely looping until an item is available. This means fewer accesses to the internal lock (and hence it is slightly faster) as compared to the non blocking version; but the main thread cannot do anything else while it is blocked. In our case, the main thread just loops back, so this is completely fine. The only reason why we had a non blocking version was to exit on `KeyboardInterrupt`, but having a timeout works quite well too.

Reply:
Thanks for explaining @neerajprad !